controllers/webhook/runcompletion_feed.go

```diff
@@ -139,19 +139,25 @@ func (rcf RunCompletionFeed) handleEvent(ctx context.Context) func(responseWrite
 			}
 
 			var runConfiguration *pipelineshub.RunConfiguration
+			var runConfigurationErr EventError
 			if eventData.RunConfigurationName != nil && eventData.RunConfigurationName.Name != "" {
-				runConfiguration, err = rcf.fetchRunConfiguration(ctx, eventData.RunConfigurationName)
-				if err != nil {
-					err.SendHttpError(responseWriter)
-					return
-				}
+				runConfiguration, runConfigurationErr = rcf.fetchRunConfiguration(ctx, eventData.RunConfigurationName)
 			}
 
 			var run *pipelineshub.Run
+			var runErr EventError
 			if eventData.RunName != nil && eventData.RunName.Name != "" {
-				run, err = rcf.fetchRun(ctx, eventData.RunName)
-				if err != nil {
-					err.SendHttpError(responseWriter)
+				run, runErr = rcf.fetchRun(ctx, eventData.RunName)
+			}
+
+			// If none are found then return error
+			if run == nil && runConfiguration == nil {
+				if runConfigurationErr != nil {
+					runConfigurationErr.SendHttpError(responseWriter)
+					return
+				}
+				if runErr != nil {
+					runErr.SendHttpError(responseWriter)
 					return
 				}
 			}
```

Comment:
Bit confused: The comment says `if none are found then return error` but that's not true, it only returns an error if it exists.

Reply:
Oh ok it means if none are found then return any errors that occurred while trying to fetch, I see...

Bit confusing but it's fine. Why do we need a comment here at all? It's just describing what the code is doing.

Reply:
ehh, was just for clarity as this guard shortcircuits the checks below